SingularityService/src/main/java/com/hubspot/singularity/resources/TaskResource.java

```diff
@@ -609,4 +613,30 @@ private SingularityTaskShellCommandRequest startShellCommand(SingularityTaskId t
     SingularityTaskId taskIdObj = getTaskIdFromStr(taskId);
     return taskManager.getTaskShellCommandUpdates(new SingularityTaskShellCommandRequestId(taskIdObj, commandName, commandTimestamp));
   }
+
+  @GET
+  @Path("/download/{httpPrefix}/{slaveHostname}/port/{port}/path/{path}")
```

Comment:
Sort of a nit, but a better API design here could be something like `GET /download?slaveHostname=x&path=y` (since the actual resource being retrieved is a `download`). The rest of our API doesn't stick to these guidelines religiously, though, so probably not a huge deal. I'll let @ssalinas weigh in as well.

Reply:
Wouldn't it be better to have them both as `PathParam`s instead of `QueryParam`s since both the slave hostname and the path are both required fields? I did change the URL a bit to better match entity class hierarchy standards. See top answer and comments on [this](https://stackoverflow.com/questions/11552248/when-to-use-queryparam-vs-pathparam) StackOverflow post.